RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c

```diff
@@ -105,5 +112,13 @@ RiscVPlatformHandOffToDxeCore (
   DEBUG ((DEBUG_INFO, "          OpenSBI Switch mode arg1: 0x%x\n", (UINTN)&OpenSbiSwitchModeContext));
   DEBUG ((DEBUG_INFO, "          OpenSBI Switch mode handler address: 0x%x\n", (UINTN)RiscVDxeIplHandoffOpenSbiHandler));
   DEBUG ((DEBUG_INFO, "          OpenSBI Switch mode to privilege 0x%x\n", PRV_S));
-  sbi_init (ThisScratch);
+  //sbi_init (ThisScratch);
+  UINTN ThisHartid = sbi_current_hartid();
+  FirmwareContext->HartSpecific[ThisHartid]->HartSwitchMode(

```

Comment:
Do we need the intermediate `RiscVDxeIplHandoffOpenSbiHandler` function or can we directly call into DXE Core?
Is it necessary to call SwitchStack?

Reply:
We still need this wrapper function coz sbi_hart_switch_mode only deals with two arguments, however we have three args to pass to DXe core entry point.